src/Simplex/Chat/Protocol.hs

```diff
@@ -589,9 +594,23 @@ msgContentTag = \case
   MCReport {} -> MCReport_
   MCUnknown {tag} -> MCUnknown_ tag
 
-data ExtMsgContent = ExtMsgContent {content :: MsgContent, file :: Maybe FileInvitation, ttl :: Maybe Int, live :: Maybe Bool}
+data ExtMsgContent = ExtMsgContent
+  { content :: MsgContent,
+    mentions :: [MemberMention],
+    file :: Maybe FileInvitation,
+    ttl :: Maybe Int,
+    live :: Maybe Bool
+  }
+  deriving (Eq, Show)
+
+-- memberName is a locally (per message) unique display name of member.
+-- Suffixes _1, _2 should be appended to make names locally unique.
```

Comment:
it seems for each added mention we'd need to scan whole list of members (not just already added mentions), to see if `<name>_<suffix>` is not already taken as just name by another member. but also what's the point of uniqueness here? mentioned clients will "know" if mention refers to them by member id, other members still won't be able to understand who's mentioned even with suffixes. for example, if group has multiple members with name "alice", naming mentions as "alice_1", "alice_2" etc. doesn't really clarify which alices are mentioned. moreover, it even makes things worse when in different message different set of alices is mentioned with same suffixes. (what I propose is to not bother with suffixes at all and just mention by display name)

Reply:
The name should be locally (per message) unique, not per member-list. The suffix will only be used if two different Alices are mentioned in the same message.